### PR TITLE
gcviewer 1.34.1 (new formula)

### PR DESCRIPTION
**Pull request by gouketsu**
The goal is to be able to install a gcviewer to be able to
check to gc.log from a JVM

### CODE REVIEW

**Comment by DomT4** [comment]
See the previous discussions about this in #39441, particularly around the `test do`. Is there nothing in terms of functionality we can test there?

File: Library/Formula/gcviewer.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Java garbage collection visualization tool"
+  homepage "https://github.com/chewiebug/GCViewer"
+  url "https://github.com/chewiebug/GCViewer/archive/1.34.1.tar.gz"
+  sha256 "e0e97a94c80be8323772dc8953d463fe8eaf60c3a1c5f212d0b575a3b5c640ba"
+
+  depends_on :java
+  depends_on "maven" => :build
+
+  def install
+    system "mvn", "-Dmaven.surefire.debug=-Duser.language=en", "clean", "install"
```
Comment (DomT4) [inline Library/Formula/gcviewer.rb]: I suspect if you're running `mvn` here you'll need to pass `ENV.java_cache` in `def install`.

**Comment by gouketsu** [comment]
Yes,

  I have seen the previous discussion, as the subject has been closed, because You have not received any answer, and I has I need it I decided to continue the work
I can’t test the functionality because it’s a JAVA IHM tools.
Functionalities are already tested during the mvn compilation in the test phase. 

I added a test section to check if the JAR with the correct version is correctly delivered.

Tomorrow I will take a look concerning the ENV.java_cache


Regards,
Jean-Bernard Damiano

**Comment by bfontaine** [comment]
> it’s a JAVA IHM tools

Did you mean “GUI” :wink:

**Comment by gouketsu** [comment]
Yes it's a GUI.

it’s a JAVA IHM tools

Did you mean “GUI” [image: :wink:]

—
Reply to this email directly or view it on GitHub
<https://github.com/Homebrew/homebrew/pull/47778#issuecomment-170605796>.

File: Library/Formula/gcviewer.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  def install
+    ENV.java_cache
+    system "mvn", "-Dmaven.surefire.debug=-Duser.language=en", "clean", "install"
+    libexec.install Dir["*"]
+    bin.write_jar_script libexec/"target/gcviewer-1.34.1.jar", "gcviewer"
+  end
+
+  test do
+    assert(File.exist?("/usr/local/Cellar/gcviewer/1.34.1/libexec/target/gcviewer-1.34.1.jar"))
```
Comment (DomT4) [inline Library/Formula/gcviewer.rb]: This won't always be `/usr/local`. You should use:
```ruby
assert File.exist?(libexec/"target/gcviewer-#{version}.jar")
```
Or such.

Comment (gouketsu) [inline Library/Formula/gcviewer.rb]: change done based on your comment

**Comment by mikemcquaid** [comment]
Does it have a `--version` or anything we can call?

**Comment by gouketsu** [comment]
No, unfortunately
By default gcviewer can be started without parameter or with a gc log file
It doesn't support any help parameter.

When it starts, it always open the interface

**Comment by mikemcquaid** [comment]
No worries :+1:

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/99daa047616e17fa35d26959bfb4436bd33beee8. Thank you for your contribution to Homebrew @gouketsu; we appreciate it! :smiley_cat:

**Pull request closed without merging**